### DISCUSSION TITLE
About sqlalchemy in long time run scripts

### DISCUSSION THREAD

**Discussion by roronoaihxn** [sqlalchemy/sqlalchemy#8424] [Usage Questions]
my scripts run in single process and mutil-thread, when my scripts receive a mqtt message from client, i need to get a scope_session,   and do some db operation according the message content, when the message done, i need to commit the message and remove the scope_session. My problem is: Can i just use a scope_func like this:
```python
def scope_func():
    return uuid.uuid4().hex
```
I can not  find any document about session use in a scripts. Can you help me, thanks!

**Comment by CaselIT** [comment]
Hi,

No, since `scope_func` will return a different value each time it's called, so each time you call the scope would get a new session even when processing the same message

>my scripts run in single process and mutil-thread

if you use multithread and a thread processes as single message at a time then the default scope function should be enough (since it should tied to the current thread)

In general the advice from sqlalchemy devs in recent years is to avoid using scrope_session and instead explicitly pass around a session

**Reply by roronoaihxn** [reply]
> Hi,
> 
> No, since `scope_func` will return a different value each time it's called, so each time you call the scope would get a new session even when processing the same message
> 
> > my scripts run in single process and mutil-thread
> 
> if you use multithread and a thread processes as single message at a time then the default scope function should be enough (since it should tied to the current thread)
> 
> In general the advice from sqlalchemy devs in recent years is to avoid using scrope_session and instead explicitly pass around a session

Thanks for your help. I want to known is the one scope_session is one mysql connection or a mysql connection from a pool?  My situation is :
 ```python

DBSession = scoped_session(sessionmaker(), scopefunc=myscopefunc) # init after all module import

def myscopefunc():
    return uuid.uuid4.hex

def on_message(message_content)
    session = DBSession()  # scope_function call only one time right? so uuid.uuid4.hex will return a fixed value when it first call, and i only call DBSession() one time. So my solution has any problem?
    session.add(xxx)  # this is same session
    session.update(xxx) # this is same session
    session.commit() # this is same session
    session.remove() # dispose this session
```
I only want to ensure the on message use the same session, every time the on_message done, i will dispose the session.

**Reply by CaselIT** [reply]
You have no need of a scoped session if you call it only once then use the session instance directly. You can just use the a sessionmaker. If the code snippet is representative of your real code my suggestion is to not use a scoped session 

Scoped session is useful only if you want to reuse the same session in multiple places without passing around the instance, in this case the scope function is called multiple times to identify which session should be returned in each call. The proposed scope function is not usable for this, since each time it returns a different id.
